### PR TITLE
Switch back to `pull_request` event for PR doc build

### PR DESCRIPTION
**Pull request by regisss**
# What does this PR do?

This PR triggers the PR doc build for the `pull_request` event instead of `pull_request_target`. `pull_request_target` was used so that the secrets needed to build Furiosa's doc can be accessed from forks (after authorization). However, this event always triggers the associated workflow from main in PRs, which is not convenient when modifying this specific workflow because we cannot see the changes.
As a workaround, this PR switches back to `pull_request` and the doc of Furiosa is not built for PRs.
Needed for #1624.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: .github/workflows/build_pr_documentation.yml [hunk 1]
```diff
@@ -42,7 +34,6 @@ jobs:
         with:
           repository: 'huggingface/optimum'
           path: optimum
-          ref: ${{ github.event.pull_request.merge_commit_sha }}
```
Comment (fxmarty) [inline .github/workflows/build_pr_documentation.yml:45]: do you mean that this SHA is main latest commit SHA instead of a potential merge commit sha?

Comment (regisss) [reply .github/workflows/build_pr_documentation.yml:45]: No, by default the ref will point to the commit SHA of the event that triggered the workflow

File: .github/workflows/build_pr_documentation.yml [hunk 2]
```diff
@@ -4,9 +4,8 @@ name: Build PR documentation
 
 on:
   workflow_dispatch:
```
Comment (echarlaix) [inline .github/workflows/build_pr_documentation.yml:6]: could be removed no ?
```suggestion
```

File: .github/workflows/build_pr_documentation.yml [hunk 3]
```diff
@@ -4,9 +4,8 @@ name: Build PR documentation
```
Comment (echarlaix) [inline .github/workflows/build_pr_documentation.yml:4]: would also remove the warning comment

**Review by echarlaix — APPROVED** [review]
Looks good, thanks @regisss

File: .github/workflows/build_pr_documentation.yml (hunk 2 above)
Comment (regisss) [reply .github/workflows/build_pr_documentation.yml:6]: I think it's good to be able to trigger the PR doc build on demand so I would rather keep it

File: .github/workflows/build_pr_documentation.yml (hunk 3 above)
Comment (regisss) [reply .github/workflows/build_pr_documentation.yml]: Done

File: .github/workflows/build_pr_documentation.yml (hunk 1 above)
Comment (fxmarty) [reply .github/workflows/build_pr_documentation.yml:45]: I don't understand. `github.event.pull_request.merge_commit_sha` should be the merge commit from a PR, is that not? Why is the documentation built against main?

Comment (regisss) [reply .github/workflows/build_pr_documentation.yml:45]: I don't know the details, but for sure it doesn't take into account the current PR changes to the workflow. See this PR for many examples: https://github.com/huggingface/optimum/pull/1624

**Pull request merged**